orchagent/portsorch.h

```diff
@@ -379,7 +380,7 @@ class PortsOrch : public Orch, public Subject
     bool setPortTpid(Port &port, sai_uint16_t tpid);
     bool setPortPvid (Port &port, sai_uint32_t pvid);
     bool getPortPvid(Port &port, sai_uint32_t &pvid);
-    bool setPortFec(Port &port, sai_port_fec_mode_t fec_mode);
+    bool setPortFec(Port &port, sai_port_fec_mode_t fec_mode, bool auto_fec);
```

Comment:
@dgsudharsan this is really confusing. Setting FEC auto is only possible while passing a FEC NONE configuration, which has nothing to do with the override action itself. Suggest considering a separate method to operate exclusively on the FEC override attribute

Reply:
@nazariig auto fec is a different mode compared to none. There are four options in the CLI - none, rs,fc, auto.
When auto is configured SAI needs to be programmed with SAI_PORT_FEC_ATTR_NONE and override is false. However since none is shared between two modes, we need a flag to differentiate whether it is auto or none mode from CLI.
Previously this API used to accept the modes as string but due to the recent refactoring the parameter is changed to sai_port_fec_mode_t . Either we undo the refactoring or introduce a new flag.

Reply:
@dgsudharsan this is what we have in the spec:
```
    /**
     * @brief FEC mode auto-negotiation override status
     *
     * If set to true, any auto-negotiated FEC mode will be
     * overridden by the value configured in SAI_PORT_ATTR_FEC_MODE
     *
     * @type bool
     * @flags CREATE_AND_SET
     * @default false
     */
    SAI_PORT_ATTR_AUTO_NEG_FEC_MODE_OVERRIDE,
```

Reply:
@nazariig  Please refer to the HLD https://github.com/sonic-net/SONiC/pull/1416.